File: schemas/media-streaming/drizzle/genres.ts
// genres: Music genre categories for classifying artists and tracks.
// See README.md for full design rationale.

import { pgTable, uuid, text, timestamp } from "drizzle-orm/pg-core";

export const genres = pgTable(
  "genres",
  {
    id: uuid("id").primaryKey().defaultRandom(),
    name: text("name").unique().notNull(),
    slug: text("slug").unique().notNull(),
    createdAt: timestamp("created_at", { withTimezone: true }).notNull().defaultNow(),
  }
);

File: schemas/media-streaming/spacetimedb/genres.rs
// genres: Music genre classifications for organizing artists and content.
// See README.md for full design rationale.

use spacetimedb::Timestamp;

#[spacetimedb::table(name = genres, public)]
pub struct Genre {
    #[primary_key]
    pub id: String, // UUID
    #[unique]
    pub name: String,
    #[unique]
    pub slug: String,
    pub created_at: Timestamp,
}
